templates/rust-starter/host/src/main.rs

```diff
@@ -2,29 +2,21 @@
 // is `multiply`, replace `METHOD_NAME_ELF` with `MULTIPLY_ELF` and replace
 // `METHOD_NAME_ID` with `MULTIPLY_ID`
 use methods::{METHOD_NAME_ELF, METHOD_NAME_ID};
-use risc0_zkvm::Prover;
+use risc0_zkvm::ExecutorEnv;
 // TODO: Uncomment the `use` line below for serialization helper functions for
 // communication with the guest
 // use risc0_zkvm::serde::{from_slice, to_vec}
 
 fn main() {
-    // Make the prover.
-    let mut prover =
-        Prover::new(METHOD_NAME_ELF).expect("Prover should be constructed from valid ELF binary");
-
-    // TODO: Implement communication with the guest here
-
-    // Run prover & generate receipt
-    let receipt = prover.run().expect(
-        "Code should be provable unless it had an error or exceeded the maximum cycle limit",
-    );
+    let env = ExecutorEnv::default();
+    let mut exec = Executor::from_elf(env, METHOD_NAME_ELF).unwrap();
+    let session = exec.run().unwrap();
```

Comment:
I would recommend keeping the .expect() with helper error messages. Provides a better first use experience for new users.

Reply:
The reason I went this way is because I think we're going to re-write these all again soon. I think a higher-level API is probably needed to reduce the boilerplate.